modern_hooks/q_object.nut

```diff
@@ -69,9 +69,22 @@
 
 		local oldParams = _oldInfos.parameters;
 		local newParams = _newInfos.parameters;
-		if (newParams[newParams.len()-1] == "..." || oldParams[oldParams.len()-1] == "...")
+		local oldHasVargv = oldParams[oldParams.len()-1] == "...";
+		local newHasVargv = newParams[newParams.len()-1] == "...";
+		if (oldHasVargv || newHasVargv)
 		{
-			// one of the functions uses vargv, do not perform validation
+			// For vargv-using functions we only want to error if the number of non-vargv
+			// parameters is different between the old and new functions
+			if (!oldHasVargv && newHasVargv)
+			{
+				if (newParams.len() - 2 > oldParams.len())
+					::Hooks.error(format("Mod %s (%s) is wrapping function %s in bb class %s with a vargv-using function but is increasing the number of non-vargv parameters from %i to %i", _q.__Mod.getID(), _q.__Mod.getName(), _key, this.buildTargetString(_q), oldParams.len() - 1, newParams.len() - 3));
+			}
+			else if (oldHasVargv && !newHasVargv)
+			{
+				if (oldParams.len() - 2 > newParams.len())
+					::Hooks.error(format("Mod %s (%s) is wrapping a vargv-using function %s in bb class %s with a non-vargv function with an increased number of non-vargv parameters (%i to %i)", _q.__Mod.getID(), _q.__Mod.getName(), _key, this.buildTargetString(_q), oldParams.len() - 3, newParams.len() - 1));
```

Comment:
Since the non-vargv version of this message is a warning, this one should be a `warn` as well for consistency.

Reply:
Sure